src/lib.rs

```diff
@@ -90,6 +90,14 @@ impl<T> StatusAnd<T> {
     }
 }
 
+impl<T: core::fmt::Debug> StatusAnd<T> {
+    /// Extract the inner value if there were no errors. If there were errors, panic.
+    pub fn unwrap(self) -> T {
```

Comment:
I think the main use case of this crate is testing softfloat libraries, e.g. compiler_builtins and libm. This method is just a convenience because the vast majority of the time you don't expect NaN and just want the inner value, but NaN would be an error so you should at least detect it.